web/src/main/java/se/inera/intyg/webcert/web/service/monitoring/MonitoringLogServiceImpl.java

```diff
@@ -241,57 +519,167 @@ public void logPrivatePractitionerTermsApproved(String userId, Personnummer pers
         boolean isAnswer) {
+        final var subject = amne != null ? amne.name() : NO_AMNE;
         if (ArendeAmne.KOMPLT == amne) {
-            logEvent(MonitoringEvent.MEDICINSKT_ARENDE_RECEIVED, intygsId, intygsTyp, unitHsaId, frageIds);
+            try (MdcCloseableMap ignored = MdcCloseableMap.builder()
+                .put(MdcLogConstants.EVENT_ACTION, toEventType(MonitoringEvent.MEDICINSKT_ARENDE_RECEIVED))
+                .put(MdcLogConstants.EVENT_TYPE, MdcLogConstants.EVENT_TYPE_CREATION)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_ID, intygsId)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_TYPE, intygsTyp)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_UNIT_ID, unitHsaId)
+                .build()
+            ) {
+                logEvent(MonitoringEvent.MEDICINSKT_ARENDE_RECEIVED, intygsId, intygsTyp, unitHsaId, frageIds);
+            }
         } else if (isAnswer) {
-            logEvent(MonitoringEvent.ARENDE_RECEIVED_ANSWER, amne != null ? amne.name() : "NO AMNE", intygsId, intygsTyp, unitHsaId);
+            try (MdcCloseableMap ignored = MdcCloseableMap.builder()
+                .put(MdcLogConstants.EVENT_ACTION, toEventType(MonitoringEvent.ARENDE_RECEIVED_ANSWER))
+                .put(MdcLogConstants.EVENT_TYPE, MdcLogConstants.EVENT_TYPE_CREATION)
+                .put(MdcLogConstants.EVENT_MESSAGE_SUBJECT, subject)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_ID, intygsId)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_TYPE, intygsTyp)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_UNIT_ID, unitHsaId)
+                .build()
+            ) {
+                logEvent(MonitoringEvent.ARENDE_RECEIVED_ANSWER, subject, intygsId, intygsTyp, unitHsaId);
+            }
         } else {
-            logEvent(MonitoringEvent.ARENDE_RECEIVED_QUESTION, amne != null ? amne.name() : "NO AMNE", intygsId, intygsTyp, unitHsaId);
+            try (MdcCloseableMap ignored = MdcCloseableMap.builder()
+                .put(MdcLogConstants.EVENT_ACTION, toEventType(MonitoringEvent.ARENDE_RECEIVED_QUESTION))
+                .put(MdcLogConstants.EVENT_TYPE, MdcLogConstants.EVENT_TYPE_CREATION)
+                .put(MdcLogConstants.EVENT_MESSAGE_SUBJECT, subject)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_ID, intygsId)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_TYPE, intygsTyp)
+                .put(MdcLogConstants.EVENT_CERTIFICATE_UNIT_ID, unitHsaId)
+                .build()
+            ) {
+                logEvent(MonitoringEvent.ARENDE_RECEIVED_QUESTION, subject, intygsId, intygsTyp, unitHsaId);
+            }
         }
     }
 
     @Override
     public void logArendeCreated(String intygsId, String intygsTyp, String unitHsaId, ArendeAmne amne, boolean isAnswer) {
+        final var subject = amne != null ? amne.name() : NO_AMNE;
```

Comment:
Would be great if we also can add event.message.id to all logs relating to an Arende was created. We need to add it as an argument.